pages/api/profile.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable camelcase */
+import { withIronSessionApiRoute } from 'iron-session/next';
+
+import httpClient from 'lib/httpClient';
+import { sessionOptions } from 'lib/session';
+
+export default withIronSessionApiRoute(profileRoute, sessionOptions);
+
+async function profileRoute(req, res) {
+  const { tokenType, accessToken } = await req.session.user;
+  const bearerToken = `${tokenType} ${accessToken}`;
+  const url = `${process.env.BACKEND_URL}/api/v1/me`;
+
+  try {
+    const {
+      data: {
+        data: { attributes },
+      },
+    } = await httpClient.get(url, {
+      headers: {
+        Authorization: bearerToken,
+      },
+    });
+
+    const { name, avatarUrl } = attributes;
+
+    res.json({ name, avatarUrl });
+  } catch (error) {
+    res.status(500).json(error);
+  }
```

Comment:
From what I understand, we fetch the user profile from the backend survey app in the server-side of the Next App, right? 
Could you share with us more about this decision? :) (and why not direct API fetching from the client).
I see some interesting points (not exposing the API key, performance with single source for the client, ...), but I'm curious to know your insights ;) 

Thanks

Reply:
Thanks @malparty , Apart from the points you mentioned, I also didn't want to expose the backend url too. I wanted to follow the [https://nextjs.org/docs/basic-features/pages#static-generation-recommended](recommended) static generation with a server side loading state followed by fetching data from client rather than fetching data at build time with `getStaticProps`. This way we are using statically generated pages, at the same time not restricting us in terms of data fetching only on build times too!

Reply:
Both solutions have advantages and drawbacks :) 
Indeed, in this current context, it might be less complex to have direct client calls to the backend IC API ;-)

Reply:
I think you are spot on! As much as the static generation is recommended, for our use case it gets a lot more complex!